apigw-sns-filter-policy-lambda-sls-py/README.md

```diff
@@ -0,0 +1,185 @@
+
+This pattern leverages the Serverless Framework to deploy a REST API via Amazon API Gateway, functioning as an Amazon SNS proxy. It incorporates SNS Filter Policy for triggering Lambda functions based on MessageBody filters. Additionally, it enables CloudWatch Logs and X-Ray tracing for both API Gateway and Lambda through the Powertools for AWS Lambda developer toolkit.
+
+Learn more about this pattern
+at [Serverless Land Patterns](https://serverlessland.com/patterns/apigw-sns-filter-policy-lambda-sls-py).
+
+**Important**: this application uses various AWS services and there are costs associated with these services after the Free
+Tier usage - please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. You are responsible for any
+AWS costs incurred. No warranty is implied in this example.
+
+## Requirements
+
+* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already
+  have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls
+  and manage AWS resources.
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+* [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) installed
+* [Python 3](https://www.python.org/downloads/) installed
+* [Serverless Framework](https://www.serverless.com/framework/docs/getting-started) installed
+* [Powertools for AWS Lambda (Python)](https://docs.powertools.aws.dev/lambda/python/latest/)
+
+## Setup and Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ```commandline
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+2. Change directory to the pattern directory:
+    ```commandline
+    cd apigw-sns-filter-policy-lambda-sls-py
+    ```
+3. Install dependencies:
+    ```commandline
+    pipenv install
+    ```
+4. Install following Serverless (sls) Framework plugins:
+    ```commandline
+    sls plugin install -n serverless-python-requirements
+    sls plugin install -n serverless-apigateway-service-proxy
+    ```
+   - `serverless-python-requirements` plugin will automatically detect `Pipfile` and pipenv installed and generate a `requirements.txt` file
+   - `serverless-apigateway-service-proxy` plugin supports the AWS service proxy integration feature of API Gateway
+
+5. Deploy the services specified in [serverless.yml](./serverless-compose.yml) by running:
+   ```commandline
+      sls deploy --verbose
+   ```
+6. The outputs from the Serverless deployment process containing the resource names, IDs and/or ARNs which are used for
+   testing.
+
+
+## How it works
+
+The high-level diagram below serves to visually represent this pattern.
+
+![apigw-sns-filter-policy-lambda-sls-py.png](./docs/apigw-sns-filter-policy-lambda-sls-py.png)
```

Comment:
I think it would be even more helpful if you update the values 2a and 2b to `foo` and `bar` respectively as in the testing examples below.